### PR TITLE
move files_sharing to IBootStrap

### PR DESCRIPTION
**Pull request by icewind1991**
reduce debug log spam

### CODE REVIEW

File: apps/files_sharing/lib/AppInfo/Application.php [hunk 1]
```diff
@@ -122,28 +87,39 @@ function () use ($c) {
...
+	}
+
+	public function boot(IBootContext $context): void {
+		$context->injectFn([$this, 'registerMountProviders']);
+		$context->injectFn([$this, 'registerEventsScripts']);
+
+		Helper::registerHooks();
+
+		Share::registerBackend('file', File::class);
+		Share::registerBackend('folder', Folder::class, 'file');
```
Comment (PVince81) [inline apps/files_sharing/lib/AppInfo/Application.php:108]: I assume that changing the order of initialization doesn't matter ?

in app.php we had these two lines run first before the Application class init code is run

File: apps/files_sharing/lib/AppInfo/Application.php [hunk 2]
```diff
@@ -122,28 +87,39 @@ function () use ($c) {
...
+		$context->registerMiddleWare(SharingCheckMiddleware::class);
+		$context->registerMiddleWare(OCSShareAPIMiddleware::class);
+		$context->registerMiddleWare(ShareInfoMiddleware::class);
 
-		$notifications->registerNotifierService(Notifier::class);
+		$context->registerCapability(Capabilities::class);
 
-		$this->registerMountProviders($mountProviderCollection);
-		$this->registerEventsScripts($dispatcher, $oldDispatcher);
-		$this->setupSharingMenus();
```
Comment (PVince81) [inline apps/files_sharing/lib/AppInfo/Application.php:133]: I didn't find this in the new code, is it obsolete ?

File: apps/files_sharing/lib/AppInfo/Application.php (hunk 1 above)
Comment (icewind1991) [reply apps/files_sharing/lib/AppInfo/Application.php:108]: I'm pretty sure it doesn't matter

File: apps/files_sharing/lib/AppInfo/Application.php (hunk 2 above)
Comment (icewind1991) [reply apps/files_sharing/lib/AppInfo/Application.php:133]: fixed the missing call to `setupSharingMenus`

**Review by ChristophWurst — APPROVED** [review]

**Comment by PVince81** [comment]
- [ ] BUG: extra nav item
 
image

**Comment by PVince81** [comment]
- [ ] BUG: missing sharing section on the left sidebar
image

@icewind1991 can you do a bit more thorough testing ? this is especially important when refactoring such deep bits of code
ideal would be to list the test cases also as checkboxes so we don't retest the same things when reviewing

**Comment by icewind1991** [comment]
Sorry, I didn't notice because the "missing image" in the navbar just shows as transparent for me, so unless I hover over the topbar the extra nav item is invisible.

Fixed.

**Comment by icewind1991** [comment]
forgot that external sharing is semi-split between 2 apps, fixed now (and improved my dev setup to allow for easier federated sharing testing)

**Comment by PVince81** [comment]
> forgot that external sharing is semi-split between 2 apps, fixed now (and improved my dev setup to allow for easier federated sharing testing)

indeed. my trick is to always pick a convoluted test scenario that is connected to the topic and is likely to break :wink:

**Review by PVince81 — APPROVED** [review]
I've tested a few scenarios like federated shares, federated group shares, link share, group share and user share, all seemed fine now :+1:

Can you also add the missing return types in the code parts you have touched ? (see code scanner comments in the PR diff)

**Review by skjnldsv — APPROVED** [review]

**Comment by skjnldsv** [comment]
`/drone/src/tests/acceptance/features/app-files-sharing.feature:249`

Is failing. php:cs too :)

**Comment by PVince81** [comment]
@icewind1991 please fix with cs:fix

**Comment by blizzz** [comment]
:ping_pong:

**Comment by szaimen** [comment]
CI is failing

**Pull request merged**